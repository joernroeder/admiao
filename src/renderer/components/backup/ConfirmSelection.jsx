import React, { useContext } from 'react'

import { ThemeContext } from '@emotion/core'
import {
	useSeedPartsDispatch,
	useSeedPartsState,
} from '../../store/SeedPartsStore'

import GridWrap from '../GridWrap'
import Cell from '../Cell'
import SubHeading from '../SubHeading'
import Text from '../Text'
import TextHeading from '../TextHeading'
import Button from '../Button'
import PieChart from '../PieChart'

const ConfirmSelection = props => {
	const dispatch = useSeedPartsDispatch()
	const { uniquePartsN, requiredPartsT } = useSeedPartsState()
	console.log(props)
	const { colors: themeColors } = useContext(ThemeContext)
	const { lightGray = '#f00', black = '#000' } = themeColors

	const onConfirmClick = () => {
		dispatch({ type: 'CONFIRM' })
	}

	const getPieChartData = () => {
		const value = 100 / uniquePartsN

		return Array(uniquePartsN)
			.fill({ value, color: lightGray })
			.map((el, index) => {
				return index < requiredPartsT ? { ...el, color: black } : el
			})
	}

	const pieChartProps = {
		width: 300,
		height: 300,
		radius: 150,
		data: getPieChartData(),
	}

	return (
		<GridWrap>
			<Cell
				gridOffset={2}
				gridColumn={3}
				style={{ textAlign: 'center' }}
				mt={7}
			>
				<PieChart {...pieChartProps} />
				<TextHeading mb={0}>
					{requiredPartsT} of {uniquePartsN}
				</TextHeading>
				<Text>Parts Required</Text>
			</Cell>
			<Cell gridOffset={1} gridColumn={5} mt={9}>
				<SubHeading mb={2}>Confirm Selection</SubHeading>
				<Text mb={9}>
					Donec ullamcorper nulla non metus auctor fringilla. Duis
					mollis, est non commodo luctus, nisi erat porttitor ligula?
				</Text>
				<Button to={'../select-parts'} showArrow={false} mr={3}>
					Nope, go back.
				</Button>
				<Button to={'../how-to-distribute'} onClick={onConfirmClick}>
					Yes, looks good!
				</Button>
			</Cell>
		</GridWrap>
	)
}

export default ConfirmSelection
