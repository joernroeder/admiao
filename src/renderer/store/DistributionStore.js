import React, { createContext, useContext, useReducer, useEffect } from 'react'

const localStorageKey = 'DistributionState'

const initialState = {
	distributionType: undefined,
	distributionIdentifier: undefined,
}

const DistributionTypes = {
	LOCAL_PRINT: 'LOCAL_PRINT',
	REMOTE_PRINT: 'REMOTE_PRINT',
	DIGITAL: 'DIGITAL',
	FILE_SYSTEM: 'FILE_SYSTEM',
}

function reducer(state, action) {
	// sets Distribution Type
	if (Object.keys(DistributionTypes).includes(action.type)) {
		return {
			...state,
			distributionType: DistributionTypes[action.type],
			distributionIdentifier: action.payload || undefined,
		}
	}

	switch (action.type) {
		case 'RESET': {
			localStorage.removeItem(localStorageKey)

			return {
				...initialState,
			}
		}

		case 'SET_IDENTIFIER': {
			return {
				...state,
				distributionIdentifier: action.payload,
			}
		}

		default:
			return state
	}
}

const DistributionStateContext = createContext(undefined)
const DistributionDispatchContext = createContext(undefined)

function DistributionProvider({ children }) {
	const [state, dispatch] = useReducer(
		reducer,
		initialState,
		initialState => {
			const previousState = localStorage.getItem(localStorageKey)

			return previousState ? JSON.parse(previousState) : initialState
		}
	)

	// store distributionType in local storage.
	// This stores the distribution type to improve the backup process.
	useEffect(() => {
		if (state.isInitialState) {
			return
		}

		// filter out empty and initial states
		if (!Object.values(state).filter(Boolean).length) {
			return
		}

		console.log(Object.values(state).filter(Boolean).length)
		localStorage.setItem(localStorageKey, JSON.stringify(state))
	}, [state])

	return (
		<DistributionStateContext.Provider value={state}>
			<DistributionDispatchContext.Provider value={dispatch}>
				{children}
			</DistributionDispatchContext.Provider>
		</DistributionStateContext.Provider>
	)
}

function useDistributionState() {
	const context = useContext(DistributionStateContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionState must be used within a DistributionProvider'
		)
	}
	return context
}

function useDistributionDispatch() {
	const context = useContext(DistributionDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionDispatch must be used within a DistributionProvider'
		)
	}
	return context
}

export {
	DistributionProvider,
	useDistributionState,
	useDistributionDispatch,
	DistributionTypes,
}
