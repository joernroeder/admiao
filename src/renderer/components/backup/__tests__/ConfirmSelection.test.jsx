import React from 'react'

import { ThemeProvider } from 'emotion-theming'
import { fireEvent, render } from '@testing-library/react'

import ConfirmSelection from '../ConfirmSelection'
import { SeedPartsProvider } from '../../../store/SeedPartsStore'

describe('Confirm Selection', () => {
	const theme = {
		colors: {},
	}

	beforeEach(() => {
		localStorage.clear()
	})

	describe('part selection should be displayed correctly', () => {
		test('correctly display 2 of 3', () => {
			const { getByText, container } = render(
				<ThemeProvider theme={theme}>
					<SeedPartsProvider>
						<ConfirmSelection />
					</SeedPartsProvider>
				</ThemeProvider>
			)

			expect(getByText('2 of 3')).toBeInTheDocument()
			expect(
				container.querySelectorAll('.pieSVG path[fill="#f00"]').length
			).toEqual(2)
			expect(
				container.querySelectorAll('.pieSVG path[fill="#ccc"]').length
			).toEqual(1)

			expect(container.firstChild).toMatchSnapshot()
		})

		test('correctly display 5 of 8', () => {
			// i know that this is an implementation detail which i am manipulating here...
			localStorage.setItem(
				'SeedPartsState',
				'{"uniquePartsN":8,"requiredPartsT":5}'
			)

			const { getByText, container } = render(
				<ThemeProvider theme={theme}>
					<SeedPartsProvider>
						<ConfirmSelection />
					</SeedPartsProvider>
				</ThemeProvider>
			)

			expect(getByText('5 of 8')).toBeInTheDocument()
			expect(
				container.querySelectorAll('.pieSVG path[fill="#f00"]').length
			).toEqual(5)
			expect(
				container.querySelectorAll('.pieSVG path[fill="#ccc"]').length
			).toEqual(3)

			expect(container.firstChild).toMatchSnapshot()
		})
	})

	test('next button should link to /how-to-distribute page', () => {
		const { history, getByText, debug } = global.renderWithRouter(
			<ThemeProvider theme={theme}>
				<SeedPartsProvider>
					<ConfirmSelection />
				</SeedPartsProvider>
			</ThemeProvider>
		)

		// headline
		expect(getByText('Confirm Selection')).toBeInTheDocument()
		expect(getByText('2 of 3')).toBeInTheDocument()

		fireEvent.click(getByText('Yes, looks good!'))

		expect(history.location.pathname).toEqual('/how-to-distribute')
	})

	test('back button should link to /select-parts page', () => {
		localStorage.setItem(
			'SeedPartsState',
			'{"uniquePartsN":8,"requiredPartsT":4}'
		)

		const { history, getByText, debug } = global.renderWithRouter(
			<ThemeProvider theme={theme}>
				<SeedPartsProvider>
					<ConfirmSelection />
				</SeedPartsProvider>
			</ThemeProvider>
		)

		// headline
		expect(getByText('Confirm Selection')).toBeInTheDocument()
		expect(getByText('4 of 8')).toBeInTheDocument()

		fireEvent.click(getByText('Nope, go back.'))

		expect(history.location.pathname).toEqual('/select-parts')
	})
})
