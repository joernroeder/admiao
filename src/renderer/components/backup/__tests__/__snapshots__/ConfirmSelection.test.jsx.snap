// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Selection part selection should be displayed correctly correctly display 2 of 3 1`] = `
.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 14px;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 256px;
  padding-left: 4px;
  padding-right: 4px;
  width: 25%;
}

.emotion-0 {
  color: black;
  margin-bottom: 0;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  -webkit-letter-spacing: 0.015em;
  -moz-letter-spacing: 0.015em;
  -ms-letter-spacing: 0.015em;
  letter-spacing: 0.015em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-1 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 9px;
  padding-left: 4px;
  padding-right: 4px;
  width: 41.66666666666667%;
}

.emotion-3 {
  color: black;
  margin-bottom: 8px;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

@media screen and (min-width:52em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

.emotion-4 {
  color: black;
  margin-bottom: 9px;
  margin-top: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-6 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: black;
  background-color: transparent;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
}

.emotion-6 .icon {
  display: inline-block;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-6:hover .icon {
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-6:hover .icon.next {
  -webkit-transform: translateX(0.4em);
  -ms-transform: translateX(0.4em);
  transform: translateX(0.4em);
}

.emotion-6:hover .icon.prev {
  -webkit-transform: translateX(-0.4em);
  -ms-transform: translateX(-0.4em);
  transform: translateX(-0.4em);
}

.emotion-6:hover .icon.flip {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-6 > span:first-of-type {
  position: relative;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-5 {
  background-color: blackTransparent;
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  -webkit-transform: skew( -14deg, 0 ) translateX( 110% ) translateY( 0% );
  -ms-transform: skew( -14deg, 0 ) translateX( 110% ) translateY( 0% );
  transform: skew( -14deg, 0 ) translateX( 110% ) translateY( 0% );
  -webkit-transition: -webkit-transform 0.25s ease-in,opacity 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in,opacity 0.25s ease-in;
  transition: transform 0.25s ease-in,opacity 0.25s ease-in;
}

*:hover > .emotion-5,
*:focus > .emotion-5 {
  opacity: 1;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transform: translateX( 1.9em ) translateY(0) skew( -14deg, 0 );
  -ms-transform: translateX( 1.9em ) translateY(0) skew( -14deg, 0 );
  transform: translateX( 1.9em ) translateY(0) skew( -14deg, 0 );
}

.emotion-8 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: white;
  background-color: black;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
}

.emotion-8 .icon {
  display: inline-block;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-8:hover .icon {
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-8:hover .icon.next {
  -webkit-transform: translateX(0.4em);
  -ms-transform: translateX(0.4em);
  transform: translateX(0.4em);
}

.emotion-8:hover .icon.prev {
  -webkit-transform: translateX(-0.4em);
  -ms-transform: translateX(-0.4em);
  transform: translateX(-0.4em);
}

.emotion-8:hover .icon.flip {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-8 > span:first-of-type {
  position: relative;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-7 {
  background-color: whiteTransparent;
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  -webkit-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -ms-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -webkit-transition: -webkit-transform 0.25s ease-in,opacity 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in,opacity 0.25s ease-in;
  transition: transform 0.25s ease-in,opacity 0.25s ease-in;
}

*:hover > .emotion-7,
*:focus > .emotion-7 {
  opacity: 1;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  -ms-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
}

<div
  class="emotion-10"
>
  <div
    class="emotion-2"
    style="text-align: center;"
  >
    <svg
      class=" pieSVG"
      viewBox="0 0 300 300"
    >
      <path
        d="M150 150 L 150 0 A 150 150 0 0 1 279.9038105676658 225.00000000000003 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 279.9038105676658 225.00000000000003 A 150 150 0 0 1 20.09618943233417 224.99999999999994 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 20.09618943233417 224.99999999999994 A 150 150 0 0 1 150.00000000000009 0 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
    </svg>
    <h3
      class="emotion-0"
      color="black"
      font-size="1"
    >
      2
       of 
      3
    </h3>
    <p
      class="emotion-1"
      color="black"
      font-size="0"
    >
      Parts Required
    </p>
  </div>
  <div
    class="emotion-9"
  >
    <h2
      class="emotion-3"
      color="black"
      font-size="1,2,2"
    >
      Confirm Selection
    </h2>
    <p
      class="emotion-4"
      color="black"
      font-size="0"
    >
      Donec ullamcorper nulla non metus auctor fringilla. Duis mollis, est non commodo luctus, nisi erat porttitor ligula?
    </p>
    <a
      class="emotion-6"
      href="/select-parts"
    >
      <span>
        <span
          class="icon prev"
        >
          ←
        </span>
         
        Nope, go back.
      </span>
      <span
        class="emotion-5"
      />
    </a>
    <a
      class="emotion-8"
      href="/how-to-distribute"
    >
      <span>
        Yes, looks good!
         
        <span
          class="icon next"
        >
          →
        </span>
      </span>
      <span
        class="emotion-7"
      />
    </a>
  </div>
</div>
`;

exports[`Confirm Selection part selection should be displayed correctly correctly display 5 of 8 1`] = `
.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 14px;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 256px;
  padding-left: 4px;
  padding-right: 4px;
  width: 25%;
}

.emotion-0 {
  color: black;
  margin-bottom: 0;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  -webkit-letter-spacing: 0.015em;
  -moz-letter-spacing: 0.015em;
  -ms-letter-spacing: 0.015em;
  letter-spacing: 0.015em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-1 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 9px;
  padding-left: 4px;
  padding-right: 4px;
  width: 41.66666666666667%;
}

.emotion-3 {
  color: black;
  margin-bottom: 8px;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

@media screen and (min-width:52em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

.emotion-4 {
  color: black;
  margin-bottom: 9px;
  margin-top: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-6 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: black;
  background-color: transparent;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
}

.emotion-6 .icon {
  display: inline-block;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-6:hover .icon {
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-6:hover .icon.next {
  -webkit-transform: translateX(0.4em);
  -ms-transform: translateX(0.4em);
  transform: translateX(0.4em);
}

.emotion-6:hover .icon.prev {
  -webkit-transform: translateX(-0.4em);
  -ms-transform: translateX(-0.4em);
  transform: translateX(-0.4em);
}

.emotion-6:hover .icon.flip {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-6 > span:first-of-type {
  position: relative;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-5 {
  background-color: blackTransparent;
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  -webkit-transform: skew( -14deg, 0 ) translateX( 110% ) translateY( 0% );
  -ms-transform: skew( -14deg, 0 ) translateX( 110% ) translateY( 0% );
  transform: skew( -14deg, 0 ) translateX( 110% ) translateY( 0% );
  -webkit-transition: -webkit-transform 0.25s ease-in,opacity 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in,opacity 0.25s ease-in;
  transition: transform 0.25s ease-in,opacity 0.25s ease-in;
}

*:hover > .emotion-5,
*:focus > .emotion-5 {
  opacity: 1;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transform: translateX( 1.9em ) translateY(0) skew( -14deg, 0 );
  -ms-transform: translateX( 1.9em ) translateY(0) skew( -14deg, 0 );
  transform: translateX( 1.9em ) translateY(0) skew( -14deg, 0 );
}

.emotion-8 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: white;
  background-color: black;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
}

.emotion-8 .icon {
  display: inline-block;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-8:hover .icon {
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-8:hover .icon.next {
  -webkit-transform: translateX(0.4em);
  -ms-transform: translateX(0.4em);
  transform: translateX(0.4em);
}

.emotion-8:hover .icon.prev {
  -webkit-transform: translateX(-0.4em);
  -ms-transform: translateX(-0.4em);
  transform: translateX(-0.4em);
}

.emotion-8:hover .icon.flip {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-8 > span:first-of-type {
  position: relative;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-7 {
  background-color: whiteTransparent;
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  -webkit-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -ms-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -webkit-transition: -webkit-transform 0.25s ease-in,opacity 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in,opacity 0.25s ease-in;
  transition: transform 0.25s ease-in,opacity 0.25s ease-in;
}

*:hover > .emotion-7,
*:focus > .emotion-7 {
  opacity: 1;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  -ms-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
}

<div
  class="emotion-10"
>
  <div
    class="emotion-2"
    style="text-align: center;"
  >
    <svg
      class=" pieSVG"
      viewBox="0 0 300 300"
    >
      <path
        d="M150 150 L 150 0 A 150 150 0 0 1 256.06601717798213 43.93398282201787 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 256.06601717798213 43.93398282201787 A 150 150 0 0 1 300 150 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 300 150 A 150 150 0 0 1 256.06601717798213 256.06601717798213 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 256.06601717798213 256.06601717798213 A 150 150 0 0 1 150.00000000000003 300 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 150.00000000000003 300 A 150 150 0 0 1 43.93398282201788 256.06601717798213 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 43.93398282201788 256.06601717798213 A 150 150 0 0 1 0 150.00000000000003 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 0 150.00000000000003 A 150 150 0 0 1 43.93398282201785 43.933982822017896 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 43.93398282201785 43.933982822017896 A 150 150 0 0 1 149.99999999999997 0 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
    </svg>
    <h3
      class="emotion-0"
      color="black"
      font-size="1"
    >
      5
       of 
      8
    </h3>
    <p
      class="emotion-1"
      color="black"
      font-size="0"
    >
      Parts Required
    </p>
  </div>
  <div
    class="emotion-9"
  >
    <h2
      class="emotion-3"
      color="black"
      font-size="1,2,2"
    >
      Confirm Selection
    </h2>
    <p
      class="emotion-4"
      color="black"
      font-size="0"
    >
      Donec ullamcorper nulla non metus auctor fringilla. Duis mollis, est non commodo luctus, nisi erat porttitor ligula?
    </p>
    <a
      class="emotion-6"
      href="/select-parts"
    >
      <span>
        <span
          class="icon prev"
        >
          ←
        </span>
         
        Nope, go back.
      </span>
      <span
        class="emotion-5"
      />
    </a>
    <a
      class="emotion-8"
      href="/how-to-distribute"
    >
      <span>
        Yes, looks good!
         
        <span
          class="icon next"
        >
          →
        </span>
      </span>
      <span
        class="emotion-7"
      />
    </a>
  </div>
</div>
`;
